src/schemas/__tests__/catch.schema.unit.spec.js

```diff
@@ -0,0 +1,361 @@
+import { createCatchSchema } from '../catch.schema.js'
+import { getSubmissionByActivityId } from '../../services/activities.service.js'
+import { isMethodInternal } from '../../services/methods.service.js'
+
+jest.mock('../../services/activities.service.js')
+jest.mock('../../services/methods.service.js')
+
+describe('catch.schema.unit', () => {
+  describe('createCatchSchema', () => {
+    const getValidPayload = () => ({
+      activity: 'activities/41001',
+      dateCaught: '2024-08-02T00:00:00+01:00',
+      species: 'species/1',
+      mass: {
+        kg: 2,
+        oz: 71,
+        type: 'METRIC'
+      },
+      method: 'methods/1',
+      released: false,
+      onlyMonthRecorded: false,
+      noDateRecorded: false
+    })
+
+    const setupMocks = ({ season = 2024, methodInternal = false } = {}) => {
+      getSubmissionByActivityId.mockResolvedValueOnce({ season })
+      isMethodInternal.mockResolvedValue(methodInternal)
+    }
+
+    describe('activity', () => {
+      it('should return an error if "activity" is missing', async () => {
+        const payload = { ...getValidPayload(), activity: undefined }
```

Comment:
This is some nitpicking but the established pattern we have in GAFL for mocking objects like this is to set them up to take override arguments, for example:

```
    const getPayload = (overrides = {}) => ({
      activity: 'activities/41001',
      dateCaught: '2024-08-02T00:00:00+01:00',
      species: 'species/1',
      mass: {
        kg: 2,
        oz: 71,
        type: 'METRIC'
      },
      method: 'methods/1',
      released: false,
      onlyMonthRecorded: false,
      noDateRecorded: false,
      ...overrides
    })
```

So that it can be called like so:

```
const payload = getValidPayload(activity: undefined)
```

What you've got here works fine too, but it might be good to switch to the established format just for the sake of consistency.